### PR TITLE
Feature/270 daily infection counter by type

### LINKED ISSUE #270: For each step - how many people were infected of each connection type
For each new person that got infected -
- Take the probability that he got infected **from each one of the submatrixes** (i.e work, home, family).
- "Roll a weighted die" to decide which connection he practically got infected through.

Example:
Say an agent got infected. I calculate the probability he'll get infected from work and get 0.1, from his family I get 0.3, and from other 0.15.
I now "roll a die" with chances that have the ratio 0.1 : 0.3 : 0.13 to get if he was infected from his work : family : other.

Then - add this to a counter of how many people got infected today from each of the connection types.
Export the data inside our main output csv.

### PR DESCRIPTION
**Pull request by rlfreiman**
closes #270. depends on PR 275!

### CODE REVIEW

File: src/corona_hakab_model/infection.py [hunk 1]
```diff
@@ -52,4 +59,18 @@ def _perform_infection(self) -> List[Agent]:
...
             new_infected.append(agent)
 
-        return new_infected
+        return new_infected, infection_methods
+
+    def _get_infection_method(self, agent_index, non_zero_columns):
+        probability_by_conn_type = []
+        for connection_type in ConnectionTypes:
+            possible_infectors = non_zero_columns[connection_type][agent_index]
+            conn_infection_probability = sum([self.manager.matrix.get(
```
Comment (tal-nc) [inline src/corona_hakab_model/infection.py]: i don't think that summing works here.
i think it should be:
..._probability = 1 - prod ([1 - self.manager.matrix.get(...) for ...])

the possibility to get infected from 3 meetings isn't the sum of the possibility to get infected in each meeting, but is 1 - the possibility to not get infected at all.

File: src/corona_hakab_model/infection.py [hunk 2]
```diff
@@ -52,4 +59,18 @@ def _perform_infection(self) -> List[Agent]:
             agent = self.manager.agents[index]
             new_infected.append(agent)
 
-        return new_infected
+        return new_infected, infection_methods
+
+    def _get_infection_method(self, agent_index, non_zero_columns):
```
Comment (tal-nc) [inline src/corona_hakab_model/infection.py]: this method ignores the medical states of the possible infectors.
should use the contagiousness vector either in selecting possible infectors, or in the probability calculation

**Review by tal-nc — CHANGES_REQUESTED** [review]

File: src/corona_hakab_model/infection.py (hunk 1 above)
Comment (rlfreiman) [reply src/corona_hakab_model/infection.py]: fixed, now each infection probability is separate

File: src/corona_hakab_model/infection.py (hunk 2 above)
Comment (rlfreiman) [reply src/corona_hakab_model/infection.py]: fixed

**Pull request closed without merging**